torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -2006,67 +2006,58 @@ Tensor qr_backward(const std::vector<torch::autograd::Variable> &grads, const Te
                                       const Tensor& A,
                                       const Tensor& Q,
                                       const Tensor& R) -> Tensor {
-    // For square and deep (tall) case we refer
-    // Walter, S.F and Lehmann, L., Algorithmic Differentiation of Linear
-    // Algebra Functions with Application in Optimum Experimental Design
-    // (Extended Version) The derivative for the QR decomposition is adapted
-    // from Eq. 42 of the above reference.
-
-    // Compute R (R')^{T}
+    // References:
+    // Matthias Seeger, Asmus Hetzel, Zhenwen Dai, Eric Meissner, Neil D. Lawrence (2018). Auto-Differentiating Linear Algebra.
```

Comment:
Why did you change the reference for the real case?
Also why is there a reference for the LQ decomposition?

Reply:
I changed the reference because a different formula is used now for the real case as well. I couldn't get the original one to work for complex, but got this one to work both for real and complex. The current formula is simpler maybe because the steps in derivation are different from the original one (from https://arxiv.org/abs/1001.1654).
LQ decomposition is the transpose of QR decomposition, in that paper they just decided to make LQ primary for them.